File: Longest_Consecutive/longest_consecutive.js
/**
 * @param {number[]} nums - The array contains integer
 * @returns {number} returns the longest consecutive
 */
var longestConsecutive = function (nums) {
	if (nums.length <= 1) {
		return nums.length;
	}
	const mapNum = new Map();

	for (const num of nums) {
		mapNum.set(num, true);
	}

	let consecutive = 0;

	for (let i = 0; i < nums.length; i++) {
		if (mapNum.has(nums[i] - 1)) {
			continue;
		}

		let sequence = 1;
		let nextSequence = nums[i] + 1;

		while (mapNum.has(nextSequence)) {
			sequence++;
			nextSequence++;
		}

		if (sequence > consecutive) {
			consecutive = sequence;
		}
	}

	return consecutive;
};

File: Longest_Consecutive/longest_consecutive.py
from typing import Dict, List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return len(nums)

        mapNum: Dict[int, bool] = {}

        for num in nums:
            mapNum[num] = True

        longest_steak = 0

        for num in mapNum.keys():
            if num - 1 not in mapNum:
                current_streak = 1
                next_num = num + 1

                while next_num in mapNum:
                    current_streak += 1
                    next_num += 1

                if current_streak > longest_steak:
                    longest_steak = current_streak

        return longest_steak
